lib/traject/annotation_config.rb

```diff
@@ -123,9 +122,15 @@
 end)
 
 to_field 'thumbnail_url_ssm', (accumulate do |_resource, context|
-  image = context.clipboard[:canvas]['images'].first['resource']['service']['@id']
-  selector = AnnotationCompatibility.new(context.clipboard[:annotation]).selector
-  region = selector['value'].sub('xywh=', '') if selector['@type'] == 'oa:FragmentSelector'
-  region ||= 'full'
-  "#{image}/#{region}/100,/0/default.jpg"
+  thumb_size = '100,'
+  image_resource = context.clipboard[:canvas]['images'].first.dig('resource', 'service', '@id')
+  if image_resource
+    selector = AnnotationCompatibility.new(context.clipboard[:annotation]).selector
+    region = selector['value'].sub('xywh=', '') if selector['@type'] == 'oa:FragmentSelector'
+    region ||= 'full'
+    "#{image_resource}/#{region}/#{thumb_size}/0/default.jpg"
+  else
+    # Case where there is no image resource. Usually a "full" rotated for a "rotation" manifest
```

Comment:
Should we need to guard against the canvas not having a thumbnail element (and/or doing some sort of dumb check to make sure the `@id` is vaguely IIIF-url-like?

Reply:
I've added a simple regex guard clause in. I also went and checked all of the "rotation" manifests. They all have the pattern we expect. Not a great way to move forward in relying on this but open to alternate solutions.